src/services/tokenService.js

```diff
@@ -20,3 +21,72 @@ export default async function getToken(
 
   return token;
 }
+
+
+
+export async function getUserToken(name) {
+  const extractUrlCode = () => {
+    const path = window.location.pathname;
+    let roomCode = null;
+    if (path.startsWith("/preview/") || path.startsWith("/meeting/")) {
+      roomCode = "";
+      for (let i = 9; i < path.length; i++) {
+        if (path[i] === "/") break;
+        roomCode += path[i]
+      }
+      if (roomCode.trim() === "") roomCode = null;
+    }
+    return roomCode;
+  }
+
+  const code = extractUrlCode();
+  axios.create({ baseURL: process.env.REACT_APP_BACKEND_API, timeout: 2000 });
+  let baseUrl;
+  const baseDomain = window.location.hostname;
+  if (baseDomain === "qa2.100ms.live") {
+    baseUrl = process.env.REACT_APP_QA_BACKEND_API;
+  }
+  else if (baseDomain === "prod2.100ms.live") {
+    baseUrl = process.env.REACT_APP_PROD_BACKEND_API;
+  }
+  else {
+    baseUrl = process.env.REACT_APP_BACKEND_API;
+  }
+  const url = baseUrl + "get-token"
+  var headers = {
+    "Content-Type": "application/json",
+    "subdomain": process.env.REACT_APP_TOKEN_GENERATION_ENDPOINT_DOMAIN
+  };
+
+  let formData = new FormData();
+  formData.append('code', code);
+  formData.append('user_id', name);
+  console.log(formData, url);
+
+  return await axios.post(url, formData, { headers: headers })
+    .then((res) => {
+      try {
+        return res.data.token;
+      } catch (err) {
+        throw Error(err)
+      }
+    })
+    .catch((err) => {
+      console.log(err);
+    })
+}
+
+export function getBackendEndpoint() {
```

Comment:
since this won't change you don't need to call this function everytime. Do something like
export const endPoint = getBackendEndPoint(); and directly use the endPoint. seems like this is kind of duplicate to the above fn. extract common logic if possible